src/openforms/formio/rendering/default.py

```diff
@@ -35,7 +36,10 @@ def is_visible(self) -> bool:
         # class.
         # In registration mode, we need to treat layout/container nodes as visible so
         # that their children are emitted too.
-        if self.mode in {RenderModes.export, RenderModes.registration}:
+        visible_modes = {RenderModes.export, RenderModes.registration}
+        if settings.DISABLE_SENDING_HIDDEN_FIELDS:
+            visible_modes.remove(RenderModes.registration)
```

Comment:
I find it odd that this line is not covered by tests, yet it is in #4142. What's going on here with the tests? :thinking:

Reply:
As discussed on slack, I added a test with a fieldset and now we hit the `ContainerMixin` as well.